include/solvers/mpm_base.tcc

```diff
@@ -939,6 +960,64 @@ void mpm::MPMBase<Tdim>::nodal_frictional_constraints(
   }
 }
 
+// Nodal pressure constraints
+template <unsigned Tdim>
+void mpm::MPMBase<Tdim>::nodal_pressure_constraints(
+    const Json& mesh_props, const std::shared_ptr<mpm::IOMesh<Tdim>>& mesh_io) {
+  try {
+    // TODO: Get the total phases
+    const unsigned Tnphases = 2;
+    // Read and assign pressure constraints
+    if (mesh_props.find("boundary_conditions") != mesh_props.end() &&
+        mesh_props["boundary_conditions"].find("pressure_constraints") !=
+            mesh_props["boundary_conditions"].end()) {
+
+      // Iterate over pressure constraints
+      for (const auto& constraints :
+           mesh_props["boundary_conditions"]["pressure_constraints"]) {
+        // Pore pressure constraint phase indice
+        unsigned constraint_phase = constraints["phase_id"];
+
+        if (constraint_phase >= Tnphases)
```

Comment:
We can't check phases in MPM class, we should check it in the nodes

Reply:
Super cool. I removed this check. It will be automatically called as the `constraints_->assign_nodal_pressure_constraints()` function is called. In short, they will reach this assertion.
https://github.com/cb-geo/mpm/blob/689d9fb43a0268c302ad81002a0acac63f8f9c5f/include/node.tcc#L178-L186